public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -228,20 +270,13 @@ export class DetectionVisualEditor extends React.Component<
   private validateData = (selections: Selection[]) => {
     const { errors } = this.state;
     selections.map((selection, selIdx) => {
-      const fieldNames = new Set<string>();
       selection.data.map((data, idx) => {
         if ('field' in data) {
           const fieldName = `field_${selIdx}_${idx}`;
           delete errors.fields[fieldName];
-          if (!data.field) {
-            errors.fields[fieldName] = 'Key name is required';
-          } else if (fieldNames.has(data.field)) {
-            errors.fields[fieldName] = 'Key name already used';
-          } else {
-            fieldNames.add(data.field);
-            if (!validateDetectionFieldName(data.field)) {
-              errors.fields[fieldName] = 'Invalid key name.';
-            }
+          if (!validateDetectionFieldName(data.field)) {
+            errors.fields[fieldName] =
+              'Invalid key name. Valid characters are a-z, A-Z, 0-9, hyphens, dots, and underscores';
```

Comment:
Is `dots` good wording here? I'm wondering whether `periods` be more recognizable to general users? 

Also nitpick, but I think we want to end the error message sentence with a period.